deepspeed/runtime/zero/partition_parameters.py

```diff
@@ -215,11 +215,16 @@ def recurse(cl):
 # Inserts _post_init_method at the end of init method
 # for all sub classes of torch.nn.Module
 class InsertPostInitMethodToModuleSubClasses(object):
-    def __init__(self, enabled=True, mem_efficient_linear=True, config=None, dtype=None):
+    def __init__(self,
+                 enabled=True,
+                 mem_efficient_linear=True,
+                 ds_config=None,
```

Comment:
Won't this be a non-back-compat change? i.e. you're proposing to rename a key in a method of a public API.

Reply:
I don't think this is public. Are you referring to [this](https://deepspeed.readthedocs.io/en/latest/zero3.html#deepspeed.zero.Init)?

Reply:
Ah, you're right, this is its super-class's init that you're proposing to change. My mistake.

Should we rename it in `zero.Init` too, and alias the old config name so that the 2 APIs are consistent?

Reply:
They are not the same type: `ds_config = DeepSpeedConfig(config, ...)`. Is this what you are referring to?

Reply:
I hear you.

This is very confusing since `zero.Init` takes config arg which is typically called `ds_config` - the file that is.

Perhaps it'd help to rename things so that they are more intuitive. I can't tell a difference between `config` and `ds_config` and `_ds_config` - they all look the same to me.

Reply:
Fair point. Will think a bit about renaming. Thanks.

Reply:
How about

- `config`, `ds_config`, _and `ds_config `=> configuration json or dict from client world into DeepSpeed API

- `engine_config `=> output of DeepSpeedConfig(`ds_config`), internal to DeepSpeed, represents deepspeed engine configuration

Reply:
But the 2 are essentially the same, except one is source and the other is the resulting object.

Proposal:

- `config_dict_or_path`: ``json file`` or dict,
- `config`: DS config object

or if `ds_` prefix is even more intuitive then:

- `ds_config_dict_or_path`: ``json file`` or dict,
- `ds_config`: DS config object

-----

as an example we use `model` and `model_name_or_path` in HF `transformers`. the former is the object, the latter is the source the latter is built from.

Reply:
I prefer the `ds_`  prefix versions, thanks. Also, thanks for sharing the HF approach, that is helpful. 

Since the `zero.Init()` API is affected, you suggested we simply alias the current `config` argument, right?

Reply:
alias and probably deprecate the old name.